pkgs/development/libraries/miopen/default.nix

```diff
@@ -0,0 +1,215 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeScript
+, pkg-config
+, cmake
+, rocm-cmake
+, rocm-runtime
+, rocm-device-libs
+, rocm-comgr
+, rocm-opencl-runtime
+, rocblas
+, rocmlir
+, hip
+, clang
+, clang-ocl
+, llvm
+, miopengemm
+, composable_kernel
+, half
+, boost
+, sqlite
+, bzip2
+, texlive ? null
+, doxygen ? null
+, sphinx ? null
+, python3Packages ? null
+, zlib ? null
+, fetchurl ? null
+, buildDocs ? false
+, buildTests ? false
+# LFS isn't working, so we will manually fetch these
+# This isn't strictly required, but is recommended
+# https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1373
+, fetchKDBs ? true
+, useOpenCL ? false
+}:
+
+assert buildDocs -> texlive != null;
+assert buildDocs -> doxygen != null;
+assert buildDocs -> sphinx != null;
+assert buildDocs -> python3Packages != null;
+assert buildTests -> zlib != null;
+assert fetchKDBs -> fetchurl != null;
+
+let
+  latex = lib.optionalAttrs buildDocs (texlive.combine {
+    inherit (texlive) scheme-small
+    latexmk
+    tex-gyre
+    fncychap
+    wrapfig
+    capt-of
+    framed
+    needspace
+    tabulary
+    varwidth
+    titlesec;
+  });
+
+  kdbs = lib.optionalAttrs fetchKDBs import ./deps.nix {
+    inherit fetchurl;
+    mirror = "https://repo.radeon.com/rocm/miopen-kernel/rel-5.0";
+  };
+in stdenv.mkDerivation (finalAttrs: {
+  pname = "miopen";
+  # We have to manually specify the repoVersion for now
+  # Find the github release or `-- MIOpen_VERSION= X.X.X` in the build log
+  repoVersion = "2.18.0";
+  rocmVersion = "5.3.3";
+  version = "${finalAttrs.repoVersion}-${finalAttrs.rocmVersion}";
+
+  outputs = [
+    "out"
+  ] ++ lib.optionals buildDocs [
+    "docs"
+  ] ++ lib.optionals buildTests [
+    "test"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "ROCmSoftwarePlatform";
+    repo = "MIOpen";
+    rev = "rocm-${finalAttrs.rocmVersion}";
+    hash = "sha256-5/JitdGJ0afzK4pGOOywRLsB3/Thc6/71sRkKIxf2Lg=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    rocm-cmake
+    hip
+    clang
+    llvm
+  ];
+
+  buildInputs = [
+    rocm-runtime
+    rocm-device-libs
+    rocm-comgr
+    rocm-opencl-runtime
+    rocblas
+    rocmlir
+    clang-ocl
+    miopengemm
+    composable_kernel
+    half
+    boost
+    sqlite
+    bzip2
+  ] ++ lib.optionals buildDocs [
+    latex
+    doxygen
+    sphinx
```

Comment:
use SphinxHook, build the documentation unconditionally and enable the doc and/or man outputs.

Reply:
Many of the other ROCm derivations I've made do the same, why use SphinxHook here?
The others also don't build the documentation unconditionally, so again, why here?

Reply:
Because I first spotted it here.

Reply:
Not having this built on hydra is rough and makes it obvious why parts of the build were made conditional.

Reply:
Okay makes sense.
It might be a little bit before I commit the new changes since I gotta rebuild everything.

Reply:
To make sure, should we still be building the documentation unconditionally?
The legal stuff like the license is already copied by default.

Reply:
Really depends on how annoying this makes it for users. I'm fine with getting this in as-is and you can look into `sphinxHook` when you find the time. No biggie.

Reply:
There are a few other derivations that build documentation so it might be easier to merge this as-is and then I'll submit a PR concerning ``sphinxHook``.
The directories are all over the place for ROCm, so I'm not entirely sure one solution will work for all of them.

Reply:
Ah right, need to fix the ``docs`` review before merge. One moment.

Reply:
After some testing, simplifying this with ``sphinxHook`` isn't a good option.
The documentation seems to depend on certain variables from [cmake](https://github.com/ROCmSoftwarePlatform/MIOpenGEMM/blob/master/doc/CMakeLists.txt).
Here's what I have done: https://github.com/Madouura/nixpkgs/commit/f4ae10e13588f4568ed462a3b5009cea10a6f21c

Reply:
We also lose the ability to _reasonably_ generate the pdf.

Reply:
Clarification: The documentation does build, but will silently error in the actual documents.
You'll get missing parts and messed up in-lines.